Attempting to drop a database which was created with `dolt_clone` fails.

Running the following script:

```sh
mkdir droptest
cd droptest
dolt init
dolt remote add pushed 'file://../pushed'
dolt push pushed main:main
dolt sql -q 'call dolt_clone("file://../pushed", "cloned"); drop database cloned;'
```

Results in:

```
error on line 1 for query  drop database cloned: database not found: cloned
WARN[0000] journal writer has already been closed (/Users/aaronson/dolttest/droptest/cloned/.dolt/noms/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv) 
panic: Close() called and reduced ref count to < 0.

goroutine 1 [running]:
github.com/dolthub/dolt/go/store/nbs.onHeapTableIndex.Close({{0x14000b4c640, 0x18, 0x4c}, {0x14000af0210, 0x8, 0x8}, {0x14000b4c658, 0x8, 0x34}, {0x14000b4c660, ...}, ...})
	/Users/aaronson/src/dolt/go/store/nbs/table_index.go:528 +0xe0
github.com/dolthub/dolt/go/store/nbs.tableReader.close({{0x14000af0220, 0x2, 0x2}, {0x107434390, 0x140008d9130}, {0x1073a81b0, 0x140000a24a0}, 0x1000})
	/Users/aaronson/src/dolt/go/store/nbs/table_reader.go:691 +0x48
github.com/dolthub/dolt/go/store/nbs.tableSet.close({0x0, 0x14000aa3c20, {0x1073b84a8, 0x1400019f140}, {0x1073a0fa0, 0x14000122340}, 0x140000b84e0})
	/Users/aaronson/src/dolt/go/store/nbs/table_set.go:277 +0x100
github.com/dolthub/dolt/go/store/nbs.(*NomsBlockStore).Close(0x1400019c160)
	/Users/aaronson/src/dolt/go/store/nbs/store.go:1307 +0x9c
github.com/dolthub/dolt/go/store/nbs.(*GenerationalNBS).Close(0x14000e6e190)
	/Users/aaronson/src/dolt/go/store/nbs/generational_chunk_store.go:229 +0x40
github.com/dolthub/dolt/go/store/types.(*ValueStore).Close(...)
	/Users/aaronson/src/dolt/go/store/types/value_store.go:799
github.com/dolthub/dolt/go/store/datas.(*database).Close(0x14000a35730?)
	/Users/aaronson/src/dolt/go/store/datas/database_common.go:267 +0x2c
github.com/dolthub/dolt/go/libraries/doltcore/dbfactory.CloseAllLocalDatabases()
	/Users/aaronson/src/dolt/go/libraries/doltcore/dbfactory/file.go:73 +0x138
main.runMain()
	/Users/aaronson/src/dolt/go/cmd/dolt/dolt.go:604 +0x2410
main.main()
	/Users/aaronson/src/dolt/go/cmd/dolt/dolt.go:238 +0x1c
```

Similarly, the results from the following point to some state in the cloned database being wrong:

```sh
mkdir droptest
cd droptest
dolt init
dolt remote add pushed 'file://../pushed'
dolt push pushed main:main
dolt sql-server &
sleep 1
dolt sql -q 'call dolt_clone("file://../pushed", "cloned"); drop database cloned;'
sleep 1
dolt sql -q 'use cloned; drop database cloned;'
sleep 1
kill %1
```

The first query does returns

```
error on line 1 for query  drop database cloned: Error 1105 (HY000): database not found: cloned
```

The second query recovers a panic in `*DoltDatabaseProvider.Close()`

And then the `sql-server` itself panics on shutdown, same as the `dolt sql` invocation.

--- a/go/libraries/doltcore/sqle/database_provider.go
+++ b/go/libraries/doltcore/sqle/database_provider.go
@@ -485,7 +485,7 @@ func (p *DoltDatabaseProvider) CloneDatabaseFromRemote(
 		return fmt.Errorf("cannot create DB, file exists at %s", dbName)
 	}
 
-	dEnv, err := p.cloneDatabaseFromRemote(ctx, dbName, remoteName, branch, remoteUrl, remoteParams)
+	err := p.cloneDatabaseFromRemote(ctx, dbName, remoteName, branch, remoteUrl, remoteParams)
 	if err != nil {
 		// Make a best effort to clean up any artifacts on disk from a failed clone
 		// before we return the error
@@ -499,7 +499,7 @@ func (p *DoltDatabaseProvider) CloneDatabaseFromRemote(
 		return err
 	}
 
-	return ConfigureReplicationDatabaseHook(ctx, p, dbName, dEnv)
+	return nil
 }
 
 // cloneDatabaseFromRemote encapsulates the inner logic for cloning a database so that if any error
@@ -510,26 +510,26 @@ func (p *DoltDatabaseProvider) cloneDatabaseFromRemote(
 	ctx *sql.Context,
 	dbName, remoteName, branch, remoteUrl string,
 	remoteParams map[string]string,
-) (*env.DoltEnv, error) {
+) error {
 	if p.remoteDialer == nil {
-		return nil, fmt.Errorf("unable to clone remote database; no remote dialer configured")
+		return fmt.Errorf("unable to clone remote database; no remote dialer configured")
 	}
 
 	// TODO: params for AWS, others that need them
 	r := env.NewRemote(remoteName, remoteUrl, nil)
 	srcDB, err := r.GetRemoteDB(ctx, types.Format_Default, p.remoteDialer)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	dEnv, err := actions.EnvForClone(ctx, srcDB.ValueReadWriter().Format(), r, dbName, p.fs, "VERSION", env.GetCurrentUserHomeDir)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	err = actions.CloneRemote(ctx, srcDB, remoteName, branch, dEnv)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	err = dEnv.RepoStateWriter().UpdateBranch(dEnv.RepoState.CWBHeadRef().GetPath(), env.BranchConfig{
@@ -537,33 +537,7 @@ func (p *DoltDatabaseProvider) cloneDatabaseFromRemote(
 		Remote: remoteName,
 	})
 
-	fkChecks, err := ctx.GetSessionVariable(ctx, "foreign_key_checks")
-	if err != nil {
-		return nil, err
-	}
-
-	opts := editor.Options{
-		Deaf: dEnv.DbEaFactory(),
-		// TODO: this doesn't seem right, why is this getting set in the constructor to the DB
-		ForeignKeyChecksDisabled: fkChecks.(int8) == 0,
-	}
-
-	db, err := NewDatabase(ctx, dbName, dEnv.DbData(), opts)
-	if err != nil {
-		return nil, err
-	}
-
-	// If we have an initialization hook, invoke it.  By default, this will
-	// be ConfigureReplicationDatabaseHook, which will setup replication
-	// for the new database if a remote url template is set.
-	err = p.InitDatabaseHook(ctx, p, dbName, dEnv)
-	if err != nil {
-		return nil, err
-	}
-
-	p.databases[formatDbMapKeyName(db.Name())] = db
-
-	return dEnv, nil
+	return p.registerNewDatabase(ctx, dbName, dEnv)
 }
 
 // DropDatabase implements the sql.MutableDatabaseProvider interface

--- a/integration-tests/bats/clone-drop.bats
+++ b/integration-tests/bats/clone-drop.bats
@@ -1,0 +1,31 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+  setup_no_dolt_init
+}
+
+teardown() {
+  stop_sql_server
+  assert_feature_version
+  teardown_common
+}
+
+@test "clone-drop: clone a database and then drop it" {
+  mkdir repo
+  cd repo
+  dolt init
+  dolt remote add pushed 'file://../pushed'
+  dolt push pushed main:main
+  dolt sql -q 'call dolt_clone("file://../pushed", "cloned"); drop database cloned;'
+}
+
+@test "clone-drop: sql-server: clone a database and then drop it" {
+  mkdir repo
+  cd repo
+  dolt init
+  dolt remote add pushed 'file://../pushed'
+  dolt push pushed main:main
+  start_sql_server
+  dolt sql -q 'call dolt_clone("file://../pushed", "cloned"); drop database cloned;'
+}

--- a/integration-tests/bats/helper/local-remote.bash
+++ b/integration-tests/bats/helper/local-remote.bash
@@ -22,6 +22,7 @@ SKIP_SERVER_TESTS=$(cat <<-EOM
 ~1pk5col-strings.bats~
 ~sql-tags.bats~
 ~empty-repo.bats~
+~clone-drops.bats~
 ~verify-constraints.bats~
 ~db-revision-specifiers.bats~
 ~ignore.bats~

